src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -33,14 +35,22 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication)
+            throws AuthenticationException {
+        User user = userService.findByEmail(authentication.getName())
+                .orElseThrow(() -> new AuthenticationException("Can't find user with this email: "
```

Comment:
consider using a more appropriate type of exception here. Please fix the same issue in all places